src/main/java/at/hannibal2/skyhanni/features/inventory/wardrobe/CustomWardrobe.kt

```diff
@@ -0,0 +1,522 @@
+import at.hannibal2.skyhanni.features.inventory.wardrobe.WardrobeAPI.inCustomWardrobe
+import at.hannibal2.skyhanni.features.inventory.wardrobe.WardrobeAPI.inWardrobe
+import at.hannibal2.skyhanni.mixins.transformers.gui.AccessorGuiContainer
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.ColorUtils.darker
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColorInt
+import at.hannibal2.skyhanni.utils.ColorUtils.withAlpha
+import at.hannibal2.skyhanni.utils.ConfigUtils.jumpToEditor
+import at.hannibal2.skyhanni.utils.DelayedRun
+import at.hannibal2.skyhanni.utils.EntityUtils.getFakePlayer
+import at.hannibal2.skyhanni.utils.InventoryUtils.clickSlot
+import at.hannibal2.skyhanni.utils.InventoryUtils.getWindowId
+import at.hannibal2.skyhanni.utils.ItemUtils.removeEnchants
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.HorizontalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.VerticalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiContainer
+import net.minecraft.client.renderer.GlStateManager
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.math.max
+import kotlin.time.Duration.Companion.milliseconds
+
+@SkyHanniModule
+object CustomWardrobe {
+
+    private val config get() = SkyHanniMod.feature.inventory.customWardrobe
+
+    private var displayRenderable: Renderable? = null
+    private var inventoryButton: Renderable? = null
+    private var editMode = false
+    private var waitingForInventoryUpdate = false
+    private var lastEditClick = SimpleTimeMark.farPast()
+
+    @SubscribeEvent
+    fun onGuiRender(event: GuiContainerEvent.BeforeDraw) {
+        if (!isEnabled() || editMode) return
+        val gui = event.gui
+        val renderable = displayRenderable ?: run {
+            update()
+            displayRenderable ?: return
+        }
+
+        // Change global wardrobe scale if its taller or wider than the screen
+        if (renderable.width > gui.width || renderable.height > gui.height) {
+            if (config.spacing.globalScale <= 1) return
+            val newScale = (config.spacing.globalScale * (0.9 / max(
+                renderable.width.toDouble() / gui.width,
+                renderable.height.toDouble() / gui.height
+            ))).toInt()
+            config.spacing.globalScale = newScale
+            ChatUtils.clickableChat(
+                "Auto-set your Global Scale in custom wardrobe, as it was too tall/wide.",
+                onClick = { config::spacing.jumpToEditor() }
+            )
+            update()
+        }
```

Comment:
split up the config globaleScale and the used globale scale. When there are split you can adjust the used globale scale instead of the config one. Since I it is annoying to set the config scale over and over, (Eg. playing on two different Resolution displays)